lib/core/test/integration/Test/Integration/Scenario/API/Wallets.hs

```diff
@@ -1460,6 +1464,127 @@ spec = do
             r <- request @ApiWallet ctx (method, endpoint) Default Empty
             expectResponseCode @IO HTTP.status405 r
             expectErrorMessage errMsg405 r
+
+    it "WALLETS_UTXO_01 - Wallet's inactivity is reflected in utxo" $ \ctx -> do
+        w <- emptyWallet ctx
+        rStat <- request @ApiUtxoStatistics ctx (getWalletUtxoEp w) Default Empty
+        expectResponseCode @IO HTTP.status200 rStat
+        expectWalletUTxO [] (snd rStat)
+
+    it "WALLETS_UTXO_02 - Sending and receiving funds updates wallet's utxo." $ \ctx -> do
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWallet ctx
+
+        --send funds
+        addrs <- listAddresses ctx wDest
+        let destination = (addrs !! 1) ^. #id
+        let coins = [13::Natural, 43, 66, 101, 1339]
+        let matrix = zip coins [1..]
+        forM_ matrix $ \(c, alreadyAbsorbed) -> do
+            let payload = Json [json|{
+                    "payments": [{
+                        "address": #{destination},
+                        "amount": {
+                            "quantity": #{c},
+                            "unit": "lovelace"
+                        }
+                    }],
+                    "passphrase": "cardano-wallet"
+                }|]
+            rTrans <- request @(ApiTransaction t) ctx (postTxEp wSrc)
+                Default payload
+            expectResponseCode @IO HTTP.status202 rTrans
+
+            rGet <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
+            let coinsSent = map fromIntegral $ take alreadyAbsorbed coins
+            verify rGet
+                [ expectEventually ctx balanceTotal (fromIntegral $ sum coinsSent)
+                , expectEventually ctx balanceAvailable (fromIntegral $ sum coinsSent)
+                ]
```

Comment:
The ~25 lines above could be replaced with:

```hs
wDest <- fixtureWalletWith ctx [13,43,66,101,1339]
```

Reply:
Probably it could be, but it'd changed the test slightly. Here the utxo is verified after each transaction, which makes the test a bit stronger.